network/common/src/main/java/org/apache/spark/network/util/ByteUnit.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.util;
+
+public enum ByteUnit {
+  BYTE (1, "Bytes"),
+  KiB (1024, "KiB"),
+  MiB (Math.pow(1024, 2), "MiB"),
+  GiB (Math.pow(1024, 3), "GiB"),
+  TiB (Math.pow(1024, 4), "TiB"),
+  PiB (Math.pow(1024, 5), "PiB");
+
+  private ByteUnit(double multiplier, String name) {
+    this.multiplier = multiplier;
+    this.name = name;
+  }
+
+  // Interpret the provided number (d) with suffix (u) as this unit type.
+  // E.g. KiB.interpret(1, MiB) interprets 1MiB as its KiB representation = 1024k
+  public long convertFrom(long d, ByteUnit u) {
+    double converted = u.toBytes(d) / multiplier;
+    if (converted > Long.MAX_VALUE)
+      throw new IllegalArgumentException("Converted value (" + converted + ") " +
+        "exceeds Long.MAX_VALUE for " + name + ". Try a larger suffix (e.g. MiB instead of KiB)");
+    return (long) converted;  
+  }
+  
+  // Convert the provided number (d) interpreted as this unit type to unit type (u). 
+  public long convertTo(long d, ByteUnit u) {
+    double converted = toBytes(d) / u.multiplier;
+    if (converted > Long.MAX_VALUE)
+      throw new IllegalArgumentException("Converted value (" + converted + ") " +
+        "exceeds Long.MAX_VALUE for " + u.name + ". Try a larger suffix (e.g. MiB instead of KiB)");
+
+    return (long) converted;
+  }
+
+  public double toBytes(long d) {
```

Comment:
As I mentioned, I do not think that using double is the correct thing. Not only the API looks weird (`toBytes` returns double while the others return `long`), but doubles do not have infinite precision.

Instead, for example, have this:

    public long convertFrom(long d, ByteUnit u) {
      return u.convertTo(d, this);
    }

    public long convertTo(long d, ByteUnit u) {
      if (multiplier > u.multiplier) {
        long ratio = multiplier / u.multiplier;
        if (Long.MAX_VALUE / ratio < d) {
          throw new IllegalArgumentException("OVERFLOW!");
        }
        return d * ratio;
      } else {
        return d / (u.multiplier / multiplier);
      }
    }